dora/core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java

```diff
@@ -605,6 +605,7 @@ public int get(PageId pageId, int pageOffset, int bytesToRead, ReadTargetBuffer
         LOG.debug("get({},pageOffset={}) fails due to page not found", pageId, pageOffset);
         return 0;
       }
+      int originalOffset = buffer.offset();
       int bytesRead =
           getPage(pageInfo, pageOffset, bytesToRead, buffer, cacheContext);
```

Comment:
I found having the logic both here and in NoExceptionCacheManager is hard to maintain for future purpose.

how about putting this offset restore logic to `LocalCacheManager.getPage(pageInfo, pageOffset, bytesToRead, buffer, cacheContext);` which is the only entry point for `pageStore.get`? in this case you don't need to patch `NoExceptionCacheManager` any more because exceptions will be caught there too.

Reply:
getPage only handle the IO exceptions and Page not found exceptions, so we still have to patch  NoExceptionCacheManager.   I don't want the getPage to swallow all the exceptions, that's why I put the logic into two places.  Another option is put the logic into NoExceptionCacheManager, if the get returns -1 or any exception, we revert the offset.  What do you think?

Reply:
> Another option is put the logic into NoExceptionCacheManager, if the get returns -1 or any exception, we revert the offset. What do you think?
this looks good to me too